### PR TITLE
URL parsing for redis

### LINKED ISSUE #50: [Feature Request] Add support for url parsing for redis.
Url parsing support for redis

### PR DESCRIPTION
**Pull request by TroyDota**
fixes #50

### CODE REVIEW

**Review by ReneWerner87 — APPROVED** [review]

**Comment by ReneWerner87** [comment]
Thanks for the effort, great work 
@TroyDota 
Please adjust the README for the redis storage and add a unittests, then i will merge it promptly

File: redis/redis.go [hunk 1]
```diff
@@ -19,12 +19,25 @@ func New(config ...Config) *Storage {
...
-	})
+
+	var options *redis.Options
+	var err error
+
+	if cfg.URL != "" {
+		options, err = redis.ParseURL(cfg.URL)
+		if err != nil {
+			panic(err)
+		}
```
Comment (renanbastos93) [inline redis/redis.go:30]: I believe that we should make it better to be idiomatic this code to compiler.
```suggestion
	if options, err = redis.ParseURL(cfg.URL); cfg.URL != "" && err != nil {
		panic(err)
	}
```

**Review by renanbastos93 — APPROVED** [review]
Awesome!

File: redis/redis.go (hunk 1 above)
Comment (TroyDota) [reply redis/redis.go:30]: Does this not, parse the url if it wasnt specified?

Comment (renanbastos93) [reply redis/redis.go:30]: Yeah, it's correct logic. it was a little suggestion, not mandatory.

Comment (TroyDota) [reply redis/redis.go:30]: no if its correct ill do it, i was just curious if this parses the url before checking if its provided.
Like the order of operations that happen here.
parse url, then check cfg.URL != ""

**Comment by ReneWerner87** [comment]
> Thanks for the effort, great work
> @TroyDota
> Please adjust the README for the redis storage and add a unittests, then i will merge it promptly

could you add something for this in the README for the redis storage and a unittest

**Review by ReneWerner87 — CHANGES_REQUESTED** [review]
could you add something for this in the README for the redis storage and a unittest

**Comment by ReneWerner87** [comment]
Test and documentation added in #64

**Comment by ReneWerner87** [comment]
your work is included in the other pull(#64) request, thanks for the effort and contribution

**Pull request closed without merging**